### ISSUE TITLE
[Pioneer][Nile][10] Camera is broken with new 8.11r1 hal.

### ISSUE THREAD

**Issue by sjllls** [sonyxperiadev/bug_tracker#574]
**Platform:**  Nile
**Device:**  Pioneer
**Kernel version:** 4.14 (Prebuilt)
**Android version:** 10
**Software binaries version:** SW_binaries_for_Xperia_Android_10.0.7.1_r1_v6a_nile.img

**Previously working on**
Has this always been a problem, or is it a new issue?
New issue, Clean and new build on 2020.5.5 with latest sources.

**Description**
Camera app reported " Can't connect to the camera"

**How to reproduce**
Build with 8.11r1 hal.

**Log**
```
01-31 11:39:16.193     0     0 W qcom,camera-flash soc: qcom,camera-flash@0: Led led_flash0 renamed to led_flash0_1 due to name collision
01-31 11:39:16.196     0     0 E CAM-SOC msm_camera_get_reg_base: 905 err: no valid mem region for device:camss_cpp
01-31 11:39:16.196     0     0 E MSM-CPP cpp_probe: 4640 optional: camss_cpp_base not present
01-31 11:39:16.200     0     0 I MSM-CPP cpp_init_hardware: 1135 CPP HW Version: 0x60010002
01-31 11:39:16.200     0     0 E MSM-CPP cpp_init_hardware: 1153 stream_cnt:0
01-31 11:39:16.201     0     0 I MSM-CPP cpp_release_hardware: 1216 cpp hw release done
01-31 11:39:16.202     0     0 E msm_vfe48_get_dual_sync_platform_data: dual-vfe-irq not supported !
01-31 11:39:16.203     0     0 E CAM-SOC msm_camera_get_reg_base: 894 err: mem resource msm-cam not found
```
`01-31 11:39:20.383     0     0 E init    : couldn't write 642 to /dev/cpuset/camera-daemon/tasks: No such file or directory`

**Additional context**
[pioneer-cam.txt](https://github.com/sonyxperiadev/bug_tracker/files/4578309/pioneer-cam.txt)
[dmesg-pioneer-cam.txt](https://github.com/sonyxperiadev/bug_tracker/files/4578310/dmesg-pioneer-cam.txt)

**Comment by kholk** [comment]
I have updated the kernel today to address this issue, which was shared with the MSM8998 Yoshino platform and I confirm that it got solved here.
Also, @MarijnS95 and @Haxk20 tested SDM636 Ganges Mermaid and showed being fine, but @oshmoun tested Pioneer and said that there was some issue with the camera being green for some reason, but I don't exclude it being his own local mess.

Please, retry and report back!
Thanks!

**Comment by sjllls** [comment]
> I have updated the kernel today to address this issue, which was shared with the MSM8998 Yoshino platform and I confirm that it got solved here.
> Also, @MarijnS95 and @Haxk20 tested SDM636 Ganges Mermaid and showed being fine, but @oshmoun tested Pioneer and said that there was some issue with the camera being green for some reason, but I don't exclude it being his own local mess.
> 
> Please, retry and report back!
> Thanks!

@kholk  I rebuilt vendor.img and boot.img.
Camera is still broken for me. After checked the log, `dual-vfe` is still not supported.
I am rebuilding the system.img now, but I think it is not necessary.
Here is log:
[pioneer-cam-log-2020-5-9.txt](https://github.com/sonyxperiadev/bug_tracker/files/4603042/pioneer-cam-log-2020-5-9.txt)
[dmesg-pioneer-cam-2020-5-9.txt](https://github.com/sonyxperiadev/bug_tracker/files/4603043/dmesg-pioneer-cam-2020-5-9.txt)

Thanks!

**Comment by kholk** [comment]
I was rechecking the defconfig.... And there's something strange, I can't see MSM_ISPIF_V2. Can you confirm that this configuration option is not getting enabled for you?

It should be enabled to get it working.

**Comment by sjllls** [comment]
> I was rechecking the defconfig.... And there's something strange, I can't see MSM_ISPIF_V2. Can you confirm that this configuration option is not getting enabled for you?
> 
> It should be enabled to get it working.

```
# CONFIG_MSM_ISP_V1 is not set
CONFIG_MSM_ISPIF=y
# CONFIG_MSM_ISPIF_V1 is not set
# CONFIG_MSM_ISPIF_V2 is not set
# CONFIG_MSM_DUAL_ISP_SYNC is not set
```

**Comment by kholk** [comment]
Writing here for the records.
After some research it turned out that we have ispif v3 on sdm6xx, so v1 and v2 shall not be selected.

The current configuration is right.

**Comment by oshmoun** [comment]
also for the record, I can not test properly.
My observations on pioneer camera can be safely ignored. Currently seeing the same issue reported here, i.e the camera is broken.

**Comment by Haxk20** [comment]
Im just confirming that camera works absolutely fine on Mermaid for me.

**Comment by Haxk20** [comment]
@sjllls Can you confirm this is fixed or still an issue with the newest OEM blobs ?

**Comment by sjllls** [comment]
> @sjllls Can you confirm this is fixed or still an issue with the newest OEM blobs ?

@Haxk20  No problem, let me try build.

**Comment by sjllls** [comment]
@Haxk20  No it is still broken.
```
[    2.177997] CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 8, cb->name: :vfe sid [0] = 3072
[    2.177997] ,
[    2.178001] CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 8, cb->name: :vfe sid [1] = 3073
[    2.178001] ,
[    2.178165] CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 8, cb->name: :vfe sid [2] = 3074
[    2.178165] ,
[    2.178277] CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 8, cb->name: :vfe sid [3] = 3075
[    2.178277] ,
[    2.178691] iommu: Adding device soc:qcom,cam_smmu:msm_cam_smmu_cb2 to group 14
[    2.178900] CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 2, cb->name: :cpp sid [0] = 2560
[    2.178900] ,
[    2.179123] iommu: Adding device soc:qcom,cam_smmu:msm_cam_smmu_cb4 to group 15
[    2.179377] CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 2, cb->name: :jpeg_enc0 sid [0] = 2048
[    2.179377] ,
[    2.179568] iommu: Adding device soc:qcom,cam_smmu:msm_cam_smmu_cb5 to group 16
[    2.179777] CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 2, cb->name: :jpeg_dma sid [0] = 2049
[    2.179777] ,
[    2.188733] msm_actuator_platform_probe:2336 msm_actuator_platform_probe: No/Error Actuator GPIOs
[    2.191087] qcom,camera-flash soc:qcom,camera-flash@0: Led led_flash0 renamed to led_flash0_1 due to name collision
[    2.194965] CAM-SOC msm_camera_get_reg_base:905 err: no valid mem region for device:camss_cpp
[    2.195099] MSM-CPP cpp_probe:4640 optional: camss_cpp_base not present
[    2.199269] MSM-CPP cpp_init_hardware:1135 CPP HW Version: 0x60010002
[    2.199392] MSM-CPP cpp_init_hardware:1153 stream_cnt:0
[    2.200320] MSM-CPP cpp_release_hardware:1216 cpp hw release done
[    2.202617] msm_vfe48_get_dual_sync_platform_data: dual-vfe-irq not supported !
[    2.203247] CAM-SOC msm_camera_get_reg_base:894 err: mem resource msm-cam not found
[    2.203307] vfe_hw_probe:fail get dual_sync
```

[xa2-dmesg-2020-5-56.txt](https://github.com/sonyxperiadev/bug_tracker/files/4680383/xa2-dmesg-2020-5-56.txt)
[xa2-cam-2020-5-56.txt](https://github.com/sonyxperiadev/bug_tracker/files/4680384/xa2-cam-2020-5-56.txt)

**Comment by thiscantbeserious** [comment]
Still happens on V9C, GCam is completely unusable - some start and show some pictures from time to time afterwards the Phone has to be restarted otherwise every GCam which was prior not crashing on opening would crash instantly. 

ROM: PixelExperience_h3113-10.0-20200714-0622-OFFICIAL.zip
Software binaries version: SW_BINARIES_FOR_XPERIA_ANDROID_10.0.7.1_R1_V9C_NILE

**Comment by MarijnS95** [comment]
Perhaps easiest to provide a diff with a known-working device, Discovery:
(Blame GitHub for not highlighting `--word-diff`)

```diff
[    [-2.429587]-]{+2.402678]+} CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 8, cb->name: :vfe sid [0] = [-3072-]{+3072\x0a,+}
[    [-2.429587] ,^M-]
[-[    2.429590]-]{+2.402681]+} CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 8, cb->name: :vfe sid [1] = [-3073^M-]
[-[    2.429590] ,-]{+3073\x0a,+}
[    [-2.429713]-]{+2.402779]+} CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 8, cb->name: :vfe sid [2] = [-3074-]{+3074\x0a,+}
[    [-2.429713] ,^M-]
[-[    2.429835]-]{+2.402834]+} CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 8, cb->name: :vfe sid [3] = [-3075^M-]
[-[    2.429835] ,-]{+3075\x0a,+}
[    [-2.430233]-]{+2.403111]+} iommu: Adding device soc:qcom,cam_smmu:msm_cam_smmu_cb2 to group [-14-]{+15+}
[    [-2.430507]-]{+2.403335]+} CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 2, cb->name: :cpp sid [0] = [-2560-]{+2560\x0a,+}
[    [-2.430507] ,-]{+2.403488] iommu: Adding device soc:qcom,cam_smmu:msm_cam_smmu_cb4 to group 16+}
[    [-2.430738]-]{+2.403668]+} CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 2, cb->name: :jpeg_enc0 sid [0] = [-2048-]{+2048\x0a,+}
[    [-2.430738] ,-]{+2.403813] iommu: Adding device soc:qcom,cam_smmu:msm_cam_smmu_cb5 to group 17+}
[    [-2.431087]-]{+2.404045]+} CAM-SMMU cam_smmu_populate_sids:2121 __debug cnt = 2, cb->name: :jpeg_dma sid [0] = [-2049-]{+2049\x0a,+}
[    [-2.431087] ,-]{+2.409948] msm_actuator_platform_probe:2336 msm_actuator_platform_probe: No/Error Actuator GPIOs+}
[    [-2.437642]-]{+2.410223]+} msm_actuator_platform_probe:2336 msm_actuator_platform_probe: No/Error Actuator GPIOs
[    [-2.438823]-]{+2.410973] msm_ois_platform_probe:996 msm_ois_platform_probe: No/Error OIS GPIO+}
{+[    2.411381]+} qcom,camera-flash soc:qcom,camera-flash@0: Led led_flash0 renamed to led_flash0_1 due to name collision
[    [-2.440918]-]{+2.411625] qcom,camera-flash soc:qcom,camera-flash@1: Led led_flash1 renamed to led_flash1_1 due to name collision+}
{+[    2.411777] qcom,camera-flash soc:qcom,camera-flash@1: Led torch-light0 renamed to torch-light0_1 due to name collision+}
{+[    2.413687]+} CAM-SOC msm_camera_get_reg_base:905 err: no valid mem region for device:camss_cpp
[    [-2.441056]-]{+2.413789]+} MSM-CPP cpp_probe:4640 optional: camss_cpp_base not present
[    [-2.445318]-]{+2.417907]+} MSM-CPP cpp_init_hardware:1135 CPP HW Version: 0x60010002
[    [-2.445398]-]{+2.418025]+} MSM-CPP cpp_init_hardware:1153 stream_cnt:0
[    [-2.446309]-]{+2.418835]+} MSM-CPP cpp_release_hardware:1216 cpp hw release done
[    [-2.447959]-]{+2.420367]+} msm_vfe48_get_dual_sync_platform_data: dual-vfe-irq not supported !
[    [-2.448547]-]{+2.420846]+} CAM-SOC msm_camera_get_reg_base:894 err: mem resource msm-cam not found
[    [-2.448613]-]{+2.420900]+} vfe_hw_probe:fail get dual_sync
[    [-2.451860]-]{+2.423687]+} __msm_jpeg_init:1591] Jpeg Device id 0
```

Identical errors, that's not where we should look...

Let's see what happens when the camera is actually used:

<details>

<summary>Log collapsed and filtered for clarity</summary>

```
[  189.779385] msm_csid_init: CSID_VERSION = 0x30050000
[  189.780281] msm_csid_irq CSID0_IRQ_STATUS_ADDR = 0x800
[  189.790150] MSM-CPP cpp_init_hardware:1135 CPP HW Version: 0x60010002
[  189.790178] MSM-CPP cpp_init_hardware:1153 stream_cnt:0
[  189.841686] msm_cci_init:1446: hw_version = 0x10060000
[  190.128788] msm_csid_irq CSID0_IRQ_STATUS_ADDR = 0x800
[  190.229921] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  190.230117] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  190.230262] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  190.230355] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  190.230498] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  190.230584] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  190.230729] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  190.234734] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  190.236012] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  190.236792] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  190.237532] msm_vfe48_set_dual_vfe_mode: dual-vfe-irq not supported !
[  190.237693] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  190.265615] ispif_process_irq: PIX0 frame id: 0
[  190.265760] ispif_process_irq: PIX0 frame id: 0
[  190.304163] ispif_process_irq: PIX0 frame id: 1
[  190.304257] ispif_process_irq: PIX0 frame id: 1
[  190.342859] ispif_process_irq: PIX0 frame id: 2
[  192.713574] msm_private_ioctl:Notifying subdevs about potential sof freeze
[  192.713753] MSM-SENSOR-INIT msm_sensor_init_subdev_ioctl:122 default
[  192.713753] 
[  192.713895] msm_csiphy_irq CSIPHY0_IRQ_STATUS_ADDR0 = 0x4
[  192.714022] msm_csiphy_irq CSIPHY0_IRQ_STATUS_ADDR1 = 0x1
[  192.714092] msm_csiphy_irq CSIPHY0_IRQ_STATUS_ADDR2 = 0x80
[  192.714218] msm_csiphy_irq CSIPHY0_IRQ_STATUS_ADDR3 = 0x20
[  192.714287] msm_csiphy_irq CSIPHY0_IRQ_STATUS_ADDR4 = 0x0
[  192.714413] msm_csiphy_irq CSIPHY0_IRQ_STATUS_ADDR5 = 0x10
[  192.714483] msm_csiphy_irq CSIPHY0_IRQ_STATUS_ADDR6 = 0x4
[  192.714607] msm_csiphy_irq CSIPHY0_IRQ_STATUS_ADDR7 = 0x0
[  192.714676] msm_csiphy_irq CSIPHY0_IRQ_STATUS_ADDR8 = 0x82
[  192.714800] msm_csiphy_irq CSIPHY0_IRQ_STATUS_ADDR9 = 0x40
[  192.714881] msm_csid_irq CSID0_IRQ_STATUS_ADDR = 0xff
[  192.742015] ispif_process_irq: PIX0 frame id: 64
[  192.742106] ispif_process_irq: PIX0 frame id: 64
[  192.780863] ispif_process_irq: PIX0 frame id: 65
[  192.781008] ispif_process_irq: PIX0 frame id: 65
[  192.819404] ispif_process_irq: PIX0 frame id: 66
[  195.224002] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.225239] msm_vfe48_clear_dual_vfe_mode: dual-vfe-irq not supported !
[  195.225636] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.225731] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.225882] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.226076] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.226181] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.226342] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.226441] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.226604] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.226705] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.226875] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.305948] msm_isp_release_stats_stream: Invalid stream handle 504, expected 0
[  195.308815] msm_isp_release_stats_stream: Invalid stream handle 301, expected 0
[  195.310879] msm_isp_release_stats_stream: Invalid stream handle 202, expected 0
[  195.311913] msm_isp_release_stats_stream: Invalid stream handle 607, expected 0
[  195.312810] msm_isp_release_stats_stream: Invalid stream handle 700, expected 0
[  195.313700] msm_isp_release_stats_stream: Invalid stream handle 803, expected 0
[  195.314585] msm_isp_release_stats_stream: Invalid stream handle 408, expected 0
[  195.374246] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.548547] msm_isp_stats_create_stream: Stats already requested
[  195.549762] msm_isp_stats_create_stream: Stats already requested
[  195.551754] msm_isp_stats_create_stream: Stats already requested
[  195.553387] msm_isp_stats_create_stream: Stats already requested
[  195.553989] msm_isp_stats_create_stream: Stats already requested
[  195.554416] msm_isp_stats_create_stream: Stats already requested
[  195.555160] msm_isp_stats_create_stream: Stats already requested
[  195.557319] msm_isp_stats_create_stream: Stats already requested
[  195.557564] msm_isp_stats_create_stream: Stats already requested
[  195.557679] msm_isp_stats_create_stream: Stats already requested
[  195.557903] msm_isp_stats_create_stream: Stats already requested
[  195.558188] msm_isp_stats_create_stream: Stats already requested
[  195.558395] msm_isp_stats_create_stream: Stats already requested
[  195.558510] msm_isp_stats_create_stream: Stats already requested
[  195.600370] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.604349] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.605955] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  195.608671] msm_vfe48_set_dual_vfe_mode: dual-vfe-irq not supported !
[  195.608767] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  198.078379] msm_private_ioctl:Notifying subdevs about potential sof freeze
[  198.078502] MSM-SENSOR-INIT msm_sensor_init_subdev_ioctl:122 default
[  198.078502] 
[  198.107487] ispif_process_irq: PIX0 frame id: 64
[  198.107595] ispif_process_irq: PIX0 frame id: 64
[  198.146356] ispif_process_irq: PIX0 frame id: 65
[  198.146457] ispif_process_irq: PIX0 frame id: 65
[  198.184703] ispif_process_irq: PIX0 frame id: 66
[  200.590824] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  200.591304] msm_vfe48_clear_dual_vfe_mode: dual-vfe-irq not supported !
[  200.591827] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  200.591985] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  200.592108] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  200.791261] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  200.960590] msm_isp_stats_create_stream: Stats already requested
[  200.962823] msm_isp_release_stats_stream: Invalid stream handle 0, expected 0
[  200.963096] msm_isp_stats_create_stream: Stats already requested
[  200.964869] msm_isp_release_stats_stream: Invalid stream handle 0, expected 0
[  200.965580] msm_isp_stats_create_stream: Stats already requested
[  200.968570] msm_isp_release_stats_stream: Invalid stream handle 0, expected 0
[  200.968971] msm_isp_stats_create_stream: Stats already requested
[  200.972345] msm_isp_release_stats_stream: Invalid stream handle 0, expected 0
[  200.972650] msm_isp_stats_create_stream: Stats already requested
[  200.974370] msm_isp_release_stats_stream: Invalid stream handle 0, expected 0
[  200.974485] msm_isp_stats_create_stream: Stats already requested
[  200.976493] msm_isp_release_stats_stream: Invalid stream handle 0, expected 0
[  200.976698] msm_isp_stats_create_stream: Stats already requested
[  200.978404] msm_isp_release_stats_stream: Invalid stream handle 0, expected 0
[  201.038210] msm_isp_axi_create_stream:63 invalid state 1 expected 0
[  201.038299] msm_isp_request_axi_stream: create stream failed
[  201.109116] msm_isp_axi_create_stream:63 invalid state 1 expected 0
[  201.109280] msm_isp_request_axi_stream: create stream failed
[  203.185494] adm_open: Force 24-bits audio for APPTYPE 0x0
[  203.302211] msm_vfe48_dual_config_irq: dual-vfe-irq not supported !
[  203.330611] MSM-CPP cpp_release_hardware:1216 cpp hw release done
```

</details>

That's not at all like Discovery:

```
[ 1250.234206] msm_csid_init: CSID_VERSION = 0x30050000
[ 1250.235595] msm_csid_irq CSID0_IRQ_STATUS_ADDR = 0x800
[ 1250.244729] MSM-CPP cpp_init_hardware:1135 CPP HW Version: 0x60010002
[ 1250.244749] MSM-CPP cpp_init_hardware:1153 stream_cnt:0
[ 1250.295744] msm_cci_init:1446: hw_version = 0x10060000
[ 1250.520352] msm_csid_irq CSID0_IRQ_STATUS_ADDR = 0x800
[ 1250.650187] ispif_process_irq: PIX0 frame id: 0
[ 1250.650247] ispif_process_irq: PIX0 frame id: 0
[ 1250.688824] ispif_process_irq: PIX0 frame id: 1
[ 1250.688884] ispif_process_irq: PIX0 frame id: 1
[ 1250.727520] ispif_process_irq: PIX0 frame id: 2
```

Looks like Pioneer is trying too hard to grab the `dual-vfe-irq`. That may or may not be a fatal issue; userspace logcat will tell us more but I'll leave diffing that for another time.

**Comment by MarijnS95** [comment]
Unfortunately the log from @sjllls are expired in many places (ie. `chatty  : uid=1047(cameraserver) CAM_MctServ_3 expire 161 lines`) so they likely don't tell the full story at all.

Lining up unexpired parts show the following doesn't occur on Discovery:

```
E msm_isp_stats_create_stream: Stats already requested
E mm-camera: <IFACE ><ERROR> 1158: iface_register_buf: iface_register_buf: cannot save vfe_fd 137 with handle 0xb0000, rc = -1
E mm-camera: <IFACE ><ERROR> 319: iface_stats_reg_buf: iface_stats_reg_buf: isp_register_buf error = -1, rc = -1
E msm_isp_release_stats_stream: Invalid stream handle 0, expected 0
```

That's proprietary code and I can only guess where to look.

@sjllls or @thiscantbeserious perhaps one of you can regenerate short, unexpired logs?

1. Unlock your device, set it to the homescreen or where-ever you can reach the camera with a single click (try something simple, AOSP camera with Cam1 API);
2. `adb logcat -c` to clear logs;
3. Start logging, the usual `adb logcat > some_file`
4. Open the camera
5. Share the logs

**Comment by Haxk20** [comment]
I do not see this issue on my Pioneer tho.
Sure the camera goes nuts after second launch but the camera here works.
@kholk fixed the dual-vfe issue I think.

**Comment by MarijnS95** [comment]
@Haxk20 https://github.com/sonyxperiadev/kernel/pull/2300 was merged 27 days ago, @thiscantbeserious reports "some magic breakage" without logs nor anything to back it up on a 10 day old build...

Let's see if we can get confirmation.

**Comment by thiscantbeserious** [comment]
I'd love to follow up on this in regard to logs - but the device is already gone back to its owner. If I get my hands on a replacement I'll follow up with it (chances are slim anytime soon). Let me add that the lack of logs and albeit stupid recommendation of a Kernel rollback was simply a lack of knowledge on my part. I'm not a Kernel Dev, noir did I have had enough time or resources of building it myself, it's something I have assigned for myself in the future (I was glaring tough the official documentation shortly). 

The issue was certainly there - and it also prevented Open Camera and other Camera Apps from working after it crashed once with GCam (Camera can't be accessed). It happened with various versions of GCam from 5.2 up to 7.4 - the only GCam that was working shortly was the Pixel related one. Until I tried to take a photo.

I cleanly flashed the rom a few times to make sure that it wasn't a temporary issue. I also double checked all various forums on XDA in regard to the Nile platform. The camera had no issues on Stock - that said. I'm afraid I didn't even try the previous build, I was only looking for Android 9 (which I didn't find, must be related to all different variants of Nile platform). So I simply gave up ... because Stock was a nightmare and the Camera a main requirement and I was out of time.

I could've chosen to keep silent about this and just move along but it is what it is. It seems like there are to few people left with pioneer, after all and to many spread across the whole Nile platform with miniscule differences here that leads to issues (again, just my totally biased, non technical impression).
